packages/token/src/common/erc2981/erc2981.cairo

```diff
@@ -109,6 +115,175 @@ pub mod ERC2981Component {
+            self: @ComponentState<TContractState>, token_id: u256
+        ) -> (ContractAddress, u128, u128) {
+            InternalImpl::token_royalty(self, token_id)
+        }
+    }
+
+    //
+    // Ownable-based admin functions
+    //
+
+    #[embeddable_as(IERC2981AdminOwnableImpl)]
+    impl ERC2981AdminOwnable<
+        TContractState,
+        +HasComponent<TContractState>,
+        +ImmutableConfig,
+        +SRC5Component::HasComponent<TContractState>,
+        impl Ownable: OwnableComponent::HasComponent<TContractState>,
+        +Drop<TContractState>,
+    > of interface::IERC2981Admin<ComponentState<TContractState>> {
+        /// Sets the royalty information that all ids in this contract will default to.
+        ///
+        /// Requirements:
+        ///
+        /// - The caller is the contract owner.
+        /// - `receiver` cannot be the zero address.
+        /// - `fee_numerator` cannot be greater than the fee denominator.
+        fn set_default_royalty(
+            ref self: ComponentState<TContractState>, receiver: ContractAddress, fee_numerator: u128
+        ) {
+            get_dep_component!(@self, Ownable).assert_only_owner();
+            InternalImpl::set_default_royalty(ref self, receiver, fee_numerator)
+        }
+
+        /// Removes default royalty information.
+        ///
+        /// Requirements:
+        ///
+        /// - The caller is the contract owner.
+        fn delete_default_royalty(ref self: ComponentState<TContractState>) {
+            get_dep_component!(@self, Ownable).assert_only_owner();
+            InternalImpl::delete_default_royalty(ref self)
+        }
+
+        /// Sets the royalty information for a specific token id that takes precedence over the
+        /// global default.
+        ///
+        /// Requirements:
+        ///
+        /// - The caller is the contract owner.
+        /// - `receiver` cannot be the zero address.
+        /// - `fee_numerator` cannot be greater than the fee denominator.
+        fn set_token_royalty(
+            ref self: ComponentState<TContractState>,
+            token_id: u256,
+            receiver: ContractAddress,
+            fee_numerator: u128
+        ) {
+            get_dep_component!(@self, Ownable).assert_only_owner();
+            InternalImpl::set_token_royalty(ref self, token_id, receiver, fee_numerator)
+        }
+
+        /// Resets royalty information for the token id back to unset.
+        ///
+        /// Requirements:
+        ///
+        /// - The caller is the contract owner.
+        fn reset_token_royalty(ref self: ComponentState<TContractState>, token_id: u256) {
+            get_dep_component!(@self, Ownable).assert_only_owner();
+            InternalImpl::reset_token_royalty(ref self, token_id)
+        }
+    }
+
+    //
+    // AccessControl-based admin functions
+    //
+
+    // Role for the admin responsible for managing royalty settings.
+    pub const ROYALTY_ADMIN_ROLE: felt252 = 'ROYALTY_ADMIN_ROLE';
```

Comment:
Btw, what's the rationale behind that? From my point of view felt-strings would be a better choice since they are easier to work with (automatically decoded in block scanners) and the only drawback of 31 char limit is not actually bad for defining a role

And as an additional minor neat thing: no collision between two different roles can happen